sig-network/link-state-proposal.md

```diff
@@ -0,0 +1,201 @@
+# VEP #2: Adding Link State Management for vNICs
+
+## Release Signoff Checklist
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+
+## Overview
+
+The purpose of this enhancement is to introduce support for link state management (up/down) in KubeVirt. This feature
+will enable VM owners to dynamically control the link state. It aligns KubeVirt with traditional virtualization platforms
+and enhances its utility in environments requiring precise network state control.
+
+By introducing link state management, KubeVirt will:
+- Provide VM-level controls to configure the link state of individual interfaces.
+- Report link state info via the VirtualMachineInterface status.
+
+This proposal outlines the necessary changes to KubeVirt, including API changes.
+
+## Motivation
+
+Currently, KubeVirt does not support modifying or reporting the link state of virtual network interfaces.
+This limitation can lead to challenges in implementing advanced networking scenarios, such as:
+- Simulating link failures for testing and debugging.
+- Managing multi-network setups, where some interfaces need to be intentionally disabled.
+
+## Goals
+
+1. Allow users to configure the link state (up/down) of individual interfaces through the KubeVirt API.
+2. Enable link state configuration at both VM creation and runtime.
+3. Report the current link state of each VM interface.
+
+## Non Goals
+
+1. Affect network plumbing on the host - the setup process will remain unchanged.
+
+## User Stories
+
+As a VM owner I would like to be able to:
```

Comment:
The pod network is special, can we get confirmation from the stakeholders if it makes sense to have the primary network be toggled like this? I mainly want to make sure there is no special expectations about this.
Disconnecting from the pod network, which acts as a control plane as well, is not that trivial IMO.

Reply:
@phoracek what do you think?